### PR TITLE
Zecale client

### PR DESCRIPTION
**Pull request by dtebbs**

### CODE REVIEW

**Comment by AntoineRondelet** [comment]
@dtebbs this branch seems to contain the changes made in #37 
Can you confirm that you forked from `aggregator-fixes`? If so, I'll change the target branch to be `aggregator-fixes` to make the review easier.

**Comment by dtebbs** [comment]
> @dtebbs this branch seems to contain the changes made in #37
> Can you confirm that you forked from aggregator-fixes? If so, I'll change the target branch to be aggregator-fixes to make the review easier.

Ooops, that's absolutely right.  I forgot to update the target. Should be changed now.  Thanks.

**Comment by AntoineRondelet** [comment]
> > @dtebbs this branch seems to contain the changes made in #37
> > Can you confirm that you forked from aggregator-fixes? If so, I'll change the target branch to be aggregator-fixes to make the review easier.
> 
> Ooops, that's absolutely right. I forgot to update the target. Should be changed now. Thanks.

No problem, thanks. That's easier that way :)

File: client/setup.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        "mypy-protobuf==1.16",
+        "flake8==3.7.8",
+        "pylint==2.4.3",
+        "click==7.0",
+        "click-default-group==1.2",
+        "grpcio==1.24",
+        "grpcio-tools==1.24",
+    ],
+    scripts=[
+        "zecale/commands/zecale",
```
Comment (AntoineRondelet) [inline client/setup.py:34]: I'd rename `commands` to `cli` to be in line with Zeth

**Comment by AntoineRondelet** [comment]
Please make sure to change the base branch to `develop` @dtebbs as #37 is now merged

File: client/Makefile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+grpc: $(API_OUTPUT)
+	@# suppress "Nothing to do for ..." warning
+	@echo -n
+
+
+syntax: $(API_OUTPUT)
+	flake8 `git ls-files '**.py'`
+	mypy -p zecale
+	mypy -p test
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply client/Makefile]: renamed to `tests`

File: client/setup.py (hunk 1 above)
Comment (dtebbs) [reply client/setup.py]: renamed to `cli`

File: scripts/ci [hunk 3]
```diff
@@ -33,6 +33,28 @@ function check_cpp() {
...
+    # location.
+    pushd depends/zeth/client
+    python3 -m venv env
+    . env/bin/activate
+    pip install --upgrade pip --progress-bar off
+    make setup
+    deactivate
+    popd
+
+    # Execute Zecale client setup from cient/README.md
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply scripts/ci]: fixed typo

File: .github/workflows/onpullrequest.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Actions for pull requests only
+name: zecale-ci-pull-request
+
+on:
+  pull_request:
+
+env:
+  MACOS_BREW_PACKAGES: "pkg-config libomp"
```
Comment (AntoineRondelet) [inline .github/workflows/onpullrequest.yml:8]: This one doesn't seem required (maybe a copy/paste residue from Zeth?)

Comment (dtebbs) [reply .github/workflows/onpullrequest.yml]: removed

Comment (AntoineRondelet) [reply .github/workflows/onpullrequest.yml]: Thanks

**Comment by AntoineRondelet** [comment]
LGTM

**Pull request merged**